docs/source/config_files.rst

```diff
@@ -0,0 +1,107 @@
+.. _config-files:
+
+Supported configuration files
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Below is a list of supported configuration files (roughly in the order of build priority).
+
+.. contents::
+   :local:
+   :depth: 1
+
+``Dockerfile``
+^^^^^^^^^^^^^^
+
+This will be treated as a regular Dockerfile and a regular Docker build will be performed.
+The presence of a Dockerfile takes priority over and ignores all other build behavior
+specified in other configuration files.
+
+In the majority of cases, providing your own Dockerfile is not necessary as the base
+images provide core functionality, compact image sizes, and efficient builds. We recommend
+trying the other configuration files before deciding to use your own Dockerfile.
+
+See the `Binder Documentation <https://mybinder.readthedocs.io/en/latest/dockerfile.html>`_ for
+best-practices with Dockerfiles or
+`our Dockerfile-rstudio example <https://github.com/binder-examples/dockerfile-rstudio>`_
+for using repo2docker to set up RStudio.
+
+``environment.yml``
+^^^^^^^^^^^^^^^^^^^
+
+This is a conda environment specification, that lets you install packages with conda.
+You can also install files from pip in your ``environment.yml`` as well.
+Our example `enviornment.yml <https://github.com/binder-examples/python-conda_pip/blob/master/environment.yml>`_
+shows how one can specify a conda enviornment for repo2docker.
+
+``requirements.txt``
+^^^^^^^^^^^^^^^^^^^^
+
+This specifies a list of Python packages that should be installed in a
+virtualenv (or conda environment). Our `requirements.txt example <https://github.com/binder-examples/requirements/blob/master/requirements.txt>`_
+on GitHub shows a typical requirements file.
+
+``REQUIRE``
+^^^^^^^^^^^
+
+This specifies a list of Julia packages. Repositories with a  ``REQUIRE`` file
+**must also contain an** ``environment.yml`` **file**.  To see an example of a
+Julia repository with ``REQUIRE`` and ``environment.yml``,
+visit `binder-examples/julia-python <https://github.com/binder-examples/julia-python>`_.
+
+``install.R``
+^^^^^^^^^^^^^
+
+This is used to install R libraries pinned to a specific snapshot on
```

Comment:
Has anyone tested that the snapshotting works? If it does we should add a comment here "to set the date of the snapshot add a `runtime.txt` link_to_runtimetxt_doc_or_example"

Reply:
It's in the later documentation I think?

Reply:
Exactly, was thinking we should forward link to it. My thinking was that if you are new to this you might not know where to look/that this is something in our docs.